### DISCUSSION TITLE
Expose custom port in kubernetes service

### DISCUSSION THREAD

**Discussion by PhiRie** [emqx/emqx#12836] [Q&A]
Hi,

we are running emqx in version 5.4.1 deployed via helm chart in our k8s cluster. I now tried to create a custom listener via emqx config values which should listen to port 8884 and accepts mqtt connection from developers which are authenticated via certificate. The configuration seems to work fine because the listener is created and visible via admin dashboard (btw we have the same config running for default port 8883 which works totally fine).

![image](https://github.com/emqx/emqx/assets/30057008/b22f0e18-9607-4c27-900e-040cf7d55c8b)

I expected that if I create a new listener and bind a port that this port will also be exposed via the k8s service. But this seems not to be the case. Do I have to add some configuration param that the port is automatically exposed. Here is my config: 

  EMQX_LISTENERS__SSL__DEVACCESS__ENABLE: "true"
  EMQX_LISTENERS__SSL__DEVACCESS__BIND: 8884
  EMQX_LISTENERS__SSL__DEVACCESS__SSL_OPTIONS__CERTFILE: <path to cert>
  EMQX_LISTENERS__SSL__DEVACCESS__SSL_OPTIONS__KEYFILE: <path to key>
  EMQX_LISTENERS__SSL__DEVACCESS__SSL_OPTIONS__CACERTFILE: <path to ca cert>
  EMQX_LISTENERS__SSL__DEVACCESS__SSL_OPTIONS__VERIFY: verify_peer
  EMQX_LISTENERS__SSL__DEVACCESS__SSL_OPTIONS__FAIL_IF_NO_PEER_CERT: true

I also checked the values.yaml of the helm chart if there is some config options for the ports and found the service section where the default ports are enabled: https://github.com/emqx/emqx/blob/master/deploy/charts/emqx/values.yaml

![image](https://github.com/emqx/emqx/assets/30057008/aeb3a996-9cbe-4085-8ad5-1527efa9e618)

Do I have to additionally add my port there e.g. devaccess: 8884 ?

**Comment by Rory-Z** [comment]
In EMQX helm chart, you can't set yourself port, I think you can try EMQX operator https://github.com/emqx/emqx-operator
